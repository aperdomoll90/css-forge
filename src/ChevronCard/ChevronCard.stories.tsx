import { Meta, StoryFn } from '@storybook/react'
import { ChevronCard } from './ChevronCard'
import { ChevronCardPropsType } from './ChevronCard.types'
import StorybookContainer from '../StorybookContainer'
import shoe from '../media/shoe.png'

export default {
  title: 'Cards',
  component: ChevronCard,
  parameters: {
    jest: ['ChevronCard.test.tsx'],
  },
  argTypes: {
    color: { options: ['red', 'blue', '#1c2942'], control: { type: 'radio' } },
    buttonColor: { options: ['red', 'blue', '#1c2942','white'], control: { type: 'radio' } },
    buttonHover: { options: ['red', 'blue', '#1c2942','white'], control: { type: 'radio' } },
    labelColor: { options: ['red', 'blue', '#1c2942','white'], control: { type: 'radio' } },
    label: { options: ['Sea', 'Forest', 'Mountain'], control: { type: 'radio' } },
    direction: { options: ['45deg', '-45deg'], control: { type: 'radio' } },
  },
} as Meta

const ChevronCardTemplate: StoryFn<ChevronCardPropsType> = args => (
  <StorybookContainer>
    <ChevronCard {...args} />
  </StorybookContainer>
)

export const ChevronCardComponent = ChevronCardTemplate.bind({})
ChevronCardComponent.args = {
  color: 'blue',
  direction: '45deg',
  label: 'Shoes for running in the moon',
  labelColor: 'red',
  buttonLabel: 'Buy Now',
  imgUrl: shoe,
  linkUrl: 'string',
  buttonColor: 'white',
  buttonHover: 'yellow',
}
