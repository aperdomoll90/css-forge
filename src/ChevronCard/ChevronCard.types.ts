export interface ChevronCardPropsType {
  color?: 'red' | undefined
  label: string
  imgUrl: string
  linkUrl?: string
}

export interface styleVariableType {
  '--color': string
  '--direction': string
}
