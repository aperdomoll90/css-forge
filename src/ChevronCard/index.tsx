export { ChevronCard } from "./ChevronCard"
export * from './ChevronCard.types'
