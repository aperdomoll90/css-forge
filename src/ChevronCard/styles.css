.chevronCard {
  position: relative;
  display: flex;
  align-items: flex-end;
  justify-content: center;
  margin: 3rem 1.2rem;
  -webkit-box-reflect: below 5px linear-gradient(transparent, transparent, #0004);
}

.chevronCard .box {
  position: relative;
  width: 15rem;
  height: 18rem;
  margin: 0 5px;
  border-radius: 20px;
  background: linear-gradient(var(--direction), var(--color) 25%, white 25%, white 50%, var(--color) 50%, var(--color) 75%, white 75%, white 100%);
  background-size: 12rem 12rem;
  animation: animatedCardBackground 1s linear infinite;
  animation-play-state: paused;
  transition: filter 1s;
  display: flex;
  justify-content: center;
  overflow: hidden;
}

.chevronCard .box:hover {
  animation-play-state: running;
}

@keyframes animatedCardBackground {
  0% {
    background-position: 0;
  }
  100% {
    background-position: 12rem;
  }
}

.chevronCard-label {
  font-size: 1.2rem;
  width: 90%;
  position: absolute;
  top: 7rem;
  left: -100%;
  background: rgba(255, 255, 255, 0.455);
  transition: 0.5s;
  opacity: 0;
  color: var(--labelColor);
}
.chevronCard:hover .chevronCard-label {
  transform: translateX(120%);
  opacity: 1;
}

.chevronCardPicture {
  width: 97%;
  position: absolute;
  bottom: 3rem;
  left: 1rem;
  transition: 0.7s;
}

.chevronCard:hover .chevronCardPicture {
  width: 75%;
  transform: translateY(3rem) translateX(3rem);
}

.chevronCard a {
  min-width: 5rem;
  padding: 0.5rem;
  border-radius: 10px;
  outline: 2px solid var(--buttonColor);
  box-shadow: 0 0 5px black;
  font-size: 1rem;
  font-weight: 900;
  text-decoration: none;
  text-shadow: 0 0 5px black;
  color: var(--buttonColor);
  letter-spacing: 0.2rem;
  z-index: 50;
  transition: 0.5s;
  opacity: 0;
  position: absolute;
  left: 2rem;
  display: flex;
  align-items: center;
  justify-content: center;
}
.chevronCard:hover a {
  transform: translateY(-19rem);
  opacity: 1;
}

.chevronCard a:hover {
  color: var(--buttonHover);
  outline: 3px solid var(--buttonHover);
  background: rgba(101, 100, 100, 0.384);
}

@media (min-width: 768px) {
  .chevronCard {
    margin: unset;
  }
  .chevronCard .box {
    width: 17rem;
    height: 23rem;
  }
}
