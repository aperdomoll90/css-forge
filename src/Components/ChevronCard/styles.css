.crewCard {
    position: relative;
    display: flex;
    align-items: flex-end;
    justify-content: center;
    margin: 3rem 1.2rem;
    -webkit-box-reflect: below 1px linear-gradient(transparent, transparent, #0004);
  }
  
  .crewCard .box {
    position: relative;
    width: 15rem;
    height: 18rem;
    margin: 0 5px;
    border-radius: 20px;
    background: linear-gradient(var(--direction), var(--color) 25%, white 25%, white 50%, var(--color) 50%, var(--color) 75%, white 75%, white 100%);
    background-size: 7.5rem 7.5rem;
    filter: grayscale(0.7);
    animation: animatedCardBackground 1s linear infinite;
    animation-play-state: paused;
    transition: filter 1s;
    display: flex;
    justify-content: center;
    /* align-items: flex-end; */
  }
  /* Opposite way */
  /* .crewCard .box:nth-child(2) {
    background: linear-gradient(135deg, var(--transcolor) 25%, var(--transWhite) 25%, var(--transWhite) 50%, var(--transcolor) 50%, var(--transcolor) 75%, var(--transWhite) 75%, var(--transWhite) 100%);
    background-size: 60px 60px;
  } */
  .crewCard .box:hover {
    animation-play-state: running;
    filter: grayscale(0);
  }
  
  @keyframes animatedCardBackground {
    0% {
      background-position: 0;
    }
    100% {
      background-position: 7.5rem;
    }
  }
  .crewCard .box p {
    width: 120%;
    height: 4rem;
    position: absolute;
    bottom: -3.8rem;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 1.3rem;
    color: white;
    text-shadow: 0.5px 0.5px rgb(44, 43, 43);
    font-weight: 600;
    transition: 0.7s;
    z-index: 50;
    opacity: 0;
    transition: 1.5s;
    background: linear-gradient( to bottom, rgba(122, 8, 8, 0.864), transparent);
    border-radius: 10px 10px 0 0;
  }
  
  .crewCard:hover .box p {
    font-size: 1.5rem;
    opacity: 1;
    transform: translateY(-.5rem);
  }
  .crewCardPicture {
    height: 97%;
    position: absolute;
    bottom: 0;
    left: 0;
    transition: height 0.7s;
    filter: grayscale(0);
  }
  
  .crewCard:hover .crewCardPicture {
    height: 115%;
  }
  
  .crewCard a {
    text-decoration: none;
    font-size: 1rem;
    color: white;
    z-index: 50;
    letter-spacing: 0.2rem;
    position: absolute;
    margin-bottom: 1rem;
    outline: 2px solid white;
    padding: 0.5rem;
    border-radius: 10px;
  }
  
  .crewCard a:hover {
    transition: 0.7s;
    color: var(--linkGray);
    outline: 2px solid var(--linkGray);
  }
  
  @media (min-width: 768px) {
    .crewCard {
      margin: unset;
    }
    .crewCard .box {
      width: 10rem;
      height: 12rem;
    }
  }
  