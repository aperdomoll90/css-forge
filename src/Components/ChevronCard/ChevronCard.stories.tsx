import { Meta, StoryFn } from '@storybook/react'
import ChevronCard, { ChevronCardPropsType } from './ChevronCard'
import StorybookContainer from '../StorybookContainer'
import ben from './media/ben.png'


export default {
  title: 'Cards',
  component: ChevronCard,
  parameters: {
    jest: ['ChevronCard.test.tsx'],
  },
  argTypes: {
    color: { options: ['red', ''], control: { type: 'radio' } },
    label: { options: ['Sea', 'Forest', 'Mountain',], control: { type: 'radio' } },
  },
} as Meta

const ChevronCardTemplate: StoryFn<ChevronCardPropsType> = (args) => (
  <StorybookContainer>
    <ChevronCard {...args} />
  </StorybookContainer>
)

export const ChevronCardComponent = ChevronCardTemplate.bind({})
ChevronCardComponent.args = {
  label: 'label',
  imgUrl: ben,
}