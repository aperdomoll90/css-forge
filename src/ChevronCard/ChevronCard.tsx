import { ChevronCardPropsType, styleVariableType } from './ChevronCard.types'
import './styles.css'

export const ChevronCard: React.FC<ChevronCardPropsType> = ({ color, label, imgUrl, linkUrl }) => {
  let styleVariable: any = color === 'red' ? { '--color': `red`, '--direction': '-45deg' } : { '--color': '#002366', '--direction': `45deg` }

  return (
    <div className='ChevronCard'>
      <div className='box' style={styleVariable}>
        <p>{label}</p>
        <img alt='profile of barber' src={imgUrl} className='ChevronCardPicture' />
      </div>
      <a target='_blank' href={linkUrl}>
        Book Now
      </a>
    </div>
  )
}
