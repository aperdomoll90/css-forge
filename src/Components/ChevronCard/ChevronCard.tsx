import './styles.css'

export interface ChevronCardPropsType {
  color?: 'red' | undefined
  label: string
  imgUrl: string
  linkUrl?: string
}

function ChevronCard({ color, label, imgUrl, linkUrl }: ChevronCardPropsType) {
  let styleVariable
  color === 'red' ? (styleVariable = { '--color': `red`, '--direction': '-45deg' }) : (styleVariable = { '--color': '#002366', '--direction': `45deg` })

  return (
    <div className='crewCard'>
      <div className='box' style={styleVariable}>
        <p>{label}</p>
        <img alt='profile of barber' src={imgUrl} className='crewCardPicture' />
      </div>
      <a target='_blank' href={linkUrl}>
        Book Now
      </a>
    </div>
  )
}

export default ChevronCard
