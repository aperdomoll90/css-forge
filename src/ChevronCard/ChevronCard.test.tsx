import { render } from '@testing-library/react'
import {ChevronCard} from './ChevronCard'
import ben from './media/ben.png'

describe('ChevronCard component', () => {
  // const { axe, toHaveNoViolations } = require('jest-axe')
  // expect.extend(toHaveNoViolations)
  const props = {
    label: 'label',
    imgUrl: ben,
  }

  const renderComponent = () => render(<ChevronCard {...props} />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
  //   it('should demonstrate ADA compliance on this component', async () => {
  //     const { container } = renderComponent()
  //     const results = await axe(container)
  //     expect(container).toHaveNoViolations()
  //   })
})
