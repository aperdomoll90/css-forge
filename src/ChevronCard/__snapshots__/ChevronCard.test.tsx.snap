// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronCard component should match snapshots 1`] = `
<div>
  <div
    class="crewCard"
  >
    <div
      class="box"
      style="--color: #002366; --direction: 45deg;"
    >
      <p>
        label
      </p>
      <img
        alt="profile of barber"
        class="crewCardPicture"
        src="ben.png"
      />
    </div>
    <a
      target="_blank"
    >
      Book Now
    </a>
  </div>
</div>
`;
